python/lsst/pipe/tasks/insertFakes.py

```diff
@@ -0,0 +1,567 @@
+                    The image into which the fake sources should be added
+        wcs : `lsst.afw.geom.skyWcs.skyWcs.SkyWcs`
+                    WCS to use to add fake sources
+
+        Returns
+        -------
+        fakeCat : `pandas.core.frame.DataFrame`
+                    The original fakeCat trimmed to the area of the image
+        """
+
+        bbox = Box2D(image.getBBox())
+        corners = bbox.getCorners()
+
+        skyCorners = wcs.pixelToSky(corners)
+        region = ConvexPolygon([s.getVector() for s in skyCorners])
+
+        def trim(row):
+            coord = SpherePoint(row[self.config.raColName], row[self.config.decColName], radians)
+            return region.contains(coord.getVector())
+
+        return fakeCat[fakeCat.apply(trim, axis=1)]
+
+    def mkFakeGalsimGalaxies(self, fakeCat, band, photoCalib, pixelScale, psf, image):
+        """Make images of fake galaxies using GalSim.
+
+        Parameters
+        ----------
+        band : `str`
+        pixelScale : `float`
+        psf : `lsst.meas.extensions.psfex.psfexPsf.PsfexPsf`
+                    The PSF information to use to make the PSF images
+        fakeCat : `pandas.core.frame.DataFrame`
+                    The catalog of fake sources to be input
+        photoCalib : `lsst.afw.image.photoCalib.PhotoCalib`
+                    Photometric calibration to be used to calibrate the fake sources
+
+        Returns
+        -------
+        galImages : `list`
+                    A list of tuples of `lsst.afw.image.exposure.exposure.ExposureF` and
+                    `lsst.afw.geom.Point2D` of their locations.
+
+        Notes
+        -----
+
+        Fake galaxies are made by combining two sersic profiles, one for the bulge and one for the disk. Each
+        component has an individual sersic index (n), a, b and position angle (PA). The combined profile is
+        then convolved with the PSF at the specified x, y position on the image.
+
+        The names of the columns in the ``fakeCat`` are configurable and are the column names from the
+        University of Washington simulations database as default. For more information see the doc strings
+        attached to the config options.
+        """
+
+        galImages = []
+
+        self.log.info("Making %d fake galaxy images" % len(fakeCat))
+
+        for (index, row) in fakeCat.iterrows():
+            xy = afwGeom.Point2D(row["x"], row["y"])
+
+            try:
+                correctedFlux = psf.computeApertureFlux(self.config.calibFluxRadius, xy)
+                psfKernel = psf.computeKernelImage(xy).getArray()
+                psfKernel /= correctedFlux
+
+            except InvalidParameterError:
+                self.log.info("Galaxy at %0.4f, %0.4f outside of image" % (row["x"], row["y"]))
+                continue
+
+            try:
+                flux = photoCalib.magnitudeToInstFlux(row[self.config.magVar % band], xy)
+            except LogicError:
+                flux = 0
```

Comment:
It only really occurs for the really vignetted ccds and it would be a pain if it didn't just move on to checking if the next object was on the good bit.